lxqtpower/lxqtpower.cpp

```diff
@@ -66,11 +68,18 @@ bool Power::doAction(Power::Action action)
 {
     foreach(PowerProvider* provider, mProviders)
     {
-        if (provider->canAction(action) &&
-            provider->doAction(action)
-           )
+        if (provider->canAction(action))
         {
-            return true;
+            if (action == PowerSuspend || action == PowerHibernate)
+            {
+                mScreenSaver.lockScreen();
+                mLoop.exec();
+            }
+            if (provider->doAction(action))
```

Comment:
Sorry for the late reaction.
Should mLoop run after doAction? I thought it was blocking.